src/AspNetCore/Src/inprocessapplication.cxx

```diff
@@ -132,6 +135,125 @@ IN_PROCESS_APPLICATION::FindDotNetFolders(
 }
 
 VOID
+IN_PROCESS_APPLICATION::SetStdOut(
+    VOID
+)
+{
+    HRESULT      hr = S_OK;
+    BOOL         fLocked = FALSE;
+    BOOL         fResult = FALSE;
+
+    if (!m_fDoneStdRedirect)
+    {
+        // Have not set stdout yet, redirect stdout to log file
+        AcquireSRWLockExclusive(&m_srwLock);
+        fLocked = TRUE;
+        if (!m_fDoneStdRedirect)
+        {
+            //
+            // best effort
+            // no need to capture the error code as nothing we can do here
+            // in case mamanged layer exits abnormally, may not be able to capture the log content as it is buffered.
+            //
+            if (!GetConsoleWindow())
+            {
+                hr = m_pConfiguration->CreateLogFile(TRUE, &m_struLogFilePath, &m_hLogFileHandle);
+                if (FAILED(hr))
+                {
+                    goto Finished;
+                }
+                //
+                // SetStdHandle works as w3wp does not have Console
+                // Current process does not have a console
+                //
+                SetStdHandle(STD_ERROR_HANDLE, m_hLogFileHandle);
+                if (m_pConfiguration->QueryStdoutLogEnabled())
+                {
+                    SetStdHandle(STD_OUTPUT_HANDLE, m_hLogFileHandle);
+                    // not work
+                    // AllocConsole()  does not help
+                    // *stdout = *m_pStdFile;
+                    // *stderr = *m_pStdFile;
+                    // _dup2(_fileno(m_pStdFile), _fileno(stdout));
+                    // _dup2(_fileno(m_pStdFile), _fileno(stderr));
+                    // this one cannot capture the process start failure
+                    // _wfreopen_s(&m_pStdFile, struLogFileName.QueryStr(), L"w", stdout);
+
+                    // Periodically flush the log content to file
+                    // Reuse the callback function already defined in SERVER_PROCESS
+                    m_Timer.InitializeTimer(SERVER_PROCESS::TimerCallback, &m_struLogFilePath, 3000, 3000);
+                }
+            }
+            else
+            {
+                if (m_pConfiguration->QueryStdoutLogEnabled())
+                {
+                    hr = m_pConfiguration->CreateLogFile(TRUE, &m_struLogFilePath, &m_hLogFileHandle);
+                    if (FAILED(hr))
+                    {
+                        goto Finished;
+                    }
+
+                    // The process has console, e.g., IIS Express scenario
+                    CloseHandle(m_hLogFileHandle);
+                    m_hLogFileHandle = INVALID_HANDLE_VALUE;
+                    if (_wfopen_s(&m_pStdFile, m_struLogFilePath.QueryStr(), L"w") == 0)
```

Comment:
Does this force all logs (including the IIS Express logs) to go to the console?

Reply:
yes. IISExpress directly write to console. Managed app also write to console. We cannot distinguish the sources

Reply:
What happened to the idea of passing a file handle to the managed process?

Reply:
Who would accept it and re-route to it? UseIISIntegration? That's too late to capture a lot of common failures.